kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/extensions/KroxyliciousExtension.java

```diff
@@ -49,17 +50,26 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
         Parameter parameter = parameterContext.getParameter();
         Class<?> type = parameter.getType();
         LOGGER.trace("test {}: Resolving parameter ({} {})", extensionContext.getUniqueId(), type.getSimpleName(), parameter.getName());
-        if (String.class.getTypeName().equals(type.getName())) {
-            if (parameter.getName().toLowerCase().contains("namespace")) {
-                return extractK8sNamespace(extensionContext);
-            }
+        if (String.class.getTypeName().equals(type.getName()) && parameter.getName().toLowerCase().contains("namespace")) {
```

Comment:
untested, but I think the gobbledygook might be better expressed as `String.class.equals(type)`

```suggestion
        if (String.class.equals(type) && parameter.getName().toLowerCase().contains("namespace")) {
```